packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,8 +10,27 @@ import { ElementError, InitError, SupportError } from './errors/index.mjs'
  *
  * @virtual
  * @template {Element} [RootElementType=HTMLElement]
+ * @template {{[key:string]: unknown}} [ConfigType={}]
  */
 export class GOVUKFrontendComponent {
```

Comment:
**question** As not all components use a config, would there be any issue having an intermetiate `ConfigurableGOVUKFrontendComponent` that handles the configuration bit? This would likely reduce the size of the final bundle if importing only non-configurable components (as the code for handling configuration would be in a class that would be dropped by tree-shaking).